packages/framework/fluid-static/src/rootDataObject.ts

```diff
@@ -190,15 +209,28 @@ class DOProviderContainerRuntimeFactory extends BaseContainerRuntimeFactory {
 
 	private readonly initialObjects: LoadableObjectKindRecord;
 
-	public constructor(schema: ContainerSchema, compatibilityMode: CompatibilityMode) {
-		const [registryEntries, sharedObjects] = parseDataObjectsFromSharedObjects(schema);
-		const rootDataObjectFactory = new DataObjectFactory(
-			"rootDO",
+	/**
+	 * Create a new instance of a container runtime factory.
+	 * @remarks
+	 * The caller is responsible for making sure that the provided root data object factory is configured
+	 * appropriately based on the schema of the container (e.g. its registry entries contain all the
+	 * DataStore/DDS types that the schema says can be constructed).
+	 *
+	 * Most scenarios probably want to use {@link createDOProviderContainerRuntimeFactory} instead,
+	 * since it can take care of constructing the root data object factory based on the schema.
+	 *
+	 * @param schema - The schema for the container
+	 * @param compatibilityMode - Compatibility mode
+	 * @param rootDataObjectFactory - A factory that can construct the root data object.
+	 */
+	public constructor(
+		schema: ContainerSchema,
+		compatibilityMode: CompatibilityMode,
+		rootDataObjectFactory: DataObjectFactory<
```

Comment:
If I understand your scenario correctly, you just want to override the sharedObjects/registryEntries rather than the whole DataObjectFactory - maybe consider taking those specifically instead?  E.g. that way there's not a secret contract of knowing the "rootDO" type.

Reply:
I get it now, forgot you need control of the actual registry object and not just the entries.